crates/sui-network/src/state_sync/mod.rs

```diff
@@ -1305,15 +1305,19 @@ where
         PeerCheckpointRequestType::Content,
     )
     .with_checkpoint(*checkpoint.sequence_number());
+    let now = tokio::time::Instant::now();
     let Some(_contents) = get_full_checkpoint_contents(peers, &store, &checkpoint, timeout).await
     else {
         // Delay completion in case of error so we don't hammer the network with retries.
         let duration = peer_heights
             .read()
             .unwrap()
             .wait_interval_when_no_peer_to_sync_content();
-        info!("retrying checkpoint sync after {:?}", duration);
-        tokio::time::sleep(duration).await;
+        if now.elapsed() < duration {
+            let duration = duration - now.elapsed();
```

Comment:
Can sleep_until work here?

Reply:
i could but then its awkward to log the amount of time we are sleeping for?

Reply:
I guess it's slightly awkward either way, you could instead log the time you're sleeping until? but up to you